android/src/main/java/io/github/elyx0/reactnativedocumentpicker/DocumentPickerModule.java

```diff
@@ -174,22 +185,41 @@ public void onShowActivityResult(int resultCode, Intent data, Promise promise) {
 		}
 	}
 
-	private WritableMap getMetadata(Uri uri) {
+	private WritableMap getMetadata(Uri uri) throws FileNotFoundException {
 		WritableMap map = Arguments.createMap();
 
-		map.putString(FIELD_URI, uri.toString());
 
 		ContentResolver contentResolver = getReactApplicationContext().getContentResolver();
 
 		map.putString(FIELD_TYPE, contentResolver.getType(uri));
 
 		Cursor cursor = contentResolver.query(uri, null, null, null, null, null);
-
 		try {
 			if (cursor != null && cursor.moveToFirst()) {
+				String fileName = "";
 				int displayNameIndex = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
 				if (!cursor.isNull(displayNameIndex)) {
-					map.putString(FIELD_NAME, cursor.getString(displayNameIndex));
+					fileName = cursor.getString(displayNameIndex);
+					map.putString(FIELD_NAME, fileName);
+				}
+
+				if (uri != null && "content".equals(uri.getScheme())) {
+					try {
+						InputStream input = DocumentPickerModule.getInputStreamForVirtualFile(contentResolver,uri,contentResolver.getType(uri));
+						File file = new File(getReactApplicationContext().getCacheDir(),fileName);
+						OutputStream output = new FileOutputStream(file);
+						byte[] buffer = new byte[4 * 1024]; // or other buffer size
+						int read;
+						while ((read = input.read(buffer)) != -1) {
+							output.write(buffer, 0, read);
+						}
+						output.flush();
+						map.putString(FIELD_URI, file.getPath());
```

Comment:
Won't this return a file path instead of a `file://` URI breaking consistency with other platforms?

Reply:
I see. This has risk if users expect to receive content uri.
Thank you for review.